### PR TITLE
Feat/create protos to autoscaling struct

### PR DESCRIPTION
**Pull request by caiordjesus**
### What?
Include new autoscaling field in proto definitions and mappers
### Why?
To be able to interact with Maestro autoscaling feature, we need to include this struct to the proto definition and request adapters.

### CODE REVIEW

File: internal/api/handlers/requestadapters/schedulers.go [hunk 1]
```diff
@@ -339,7 +402,7 @@ func fromApiContainerEnvironments(apiEnvironments []*api.ContainerEnvironment) [
 }
 
 func fromApiForwarders(apiForwarders []*api.Forwarder) []*forwarder.Forwarder {
-	var forwarders []*forwarder.Forwarder
+	forwarders := make([]*forwarder.Forwarder, 0)
```
Comment (caiordjesus) [inline internal/api/handlers/requestadapters/schedulers.go:405]: Refactor: This was leading to failing when trying to patch forwarders list to empty list

File: internal/api/handlers/requestadapters/schedulers.go [hunk 2]
```diff
@@ -58,14 +61,53 @@ func FromApiPatchSchedulerRequestToChangeMap(request *api.PatchSchedulerRequest)
...
+			return map[string]interface{}{}, err
+		}
+		changeMap[patch_scheduler.LabelAutoscalingPolicy] = autoscalingPolicy
+	}
+
+	return changeMap, nil
 }
 
 func FromApiCreateSchedulerRequestToEntity(request *api.CreateSchedulerRequest) (*entities.Scheduler, error) {
+	schedulerAutoscaling, err := fromApiAutoscaling(request.GetAutoscaling())
```
Comment (arthur29) [inline internal/api/handlers/requestadapters/schedulers.go:106]: In this case I think that Autoscaling could be nil, right?

File: internal/core/services/scheduler_manager/patch_scheduler/patch_scheduler.go [hunk 3]
```diff
@@ -186,3 +211,40 @@ func patchContainers(containers []game_room.Container, patchSlice []map[string]i
 
 	return containers, nil
 }
+
+func patchAutoscaling(scheduler *entities.Scheduler, patchMap map[string]interface{}) error {
+	if scheduler.Autoscaling == nil {
+		scheduler.Autoscaling = &autoscaling.Autoscaling{}
+	}
+
+	if unconvertedEnabled, ok := patchMap[LabelAutoscalingEnabled]; ok {
```
[the remark this answers is not in the document]
Comment (arthur29) [reply internal/core/services/scheduler_manager/patch_scheduler/patch_scheduler.go:220]: This comment applies to the next `unconvertedMin` and `unconvertedMax`, as well

**Review by guilhermocc — APPROVED** [review]

File: internal/api/handlers/requestadapters/schedulers.go [hunk 4]
```diff
@@ -281,6 +331,45 @@ func fromApiSpec(apiSpec *api.Spec) *game_room.Spec {
...
+
+		roomOccupancy := params.GetRoomOccupancy()
+		if roomOccupancy == nil {
+			return autoscaling.Policy{}, errors.New("autoscaling policy type roomOccupancy without roomOccupancy params")
+		}
+
+		readyTarget := roomOccupancy.GetReadyTarget()
+		if readyTarget == 0 {
+			return autoscaling.Policy{}, errors.New("autoscaling policy type roomOccupancy should have a non-zero ready target")
+		}
```
[the remark this answers is not in the document]
Comment (guilhermocc) [reply internal/api/handlers/requestadapters/schedulers.go:350]: Yeah, the function autoscaling.NewAutoscaling(...) already applies those validations

File: internal/api/handlers/requestadapters/schedulers.go (hunk 2 above)
Comment (caiordjesus) [reply internal/api/handlers/requestadapters/schedulers.go:106]: Yes... Inside the fromApiAutoscaling method, we check if it is nil

**Review by guilhermocc — APPROVED** [review]

File: internal/api/handlers/requestadapters/schedulers.go [hunk 5]
```diff
@@ -398,6 +482,38 @@ func getSpec(spec game_room.Spec) *api.Spec {
...
+func getPolicyParameters(parameters autoscaling.PolicyParameters) *api.PolicyParameters {
+	return &api.PolicyParameters{
+		RoomOccupancy: getRoomOccupancy(parameters.RoomOccupancy),
+	}
+}
+
+func getRoomOccupancy(roomOccupancyParameters *autoscaling.RoomOccupancyParams) *api.RoomOccupancy {
+	return &api.RoomOccupancy{
+		ReadyTarget: float32(roomOccupancyParameters.ReadyTarget),
+	}
```
Comment (arthur29) [inline internal/api/handlers/requestadapters/schedulers.go:514]:
```suggestion
        if roomOccupancyParameters == nil {
            return nil
        }
        
	return &api.RoomOccupancy{
		ReadyTarget: float32(roomOccupancyParameters.ReadyTarget),
	}
```

File: internal/api/handlers/requestadapters/schedulers.go [hunk 6]
```diff
@@ -398,6 +482,41 @@ func getSpec(spec game_room.Spec) *api.Spec {
...
+
+	return nil
+}
+
+func getAutoscalingPolicy(autoscalingPolicy autoscaling.Policy) *api.AutoscalingPolicy {
+	return &api.AutoscalingPolicy{
+		Type:       string(autoscalingPolicy.Type),
+		Parameters: getPolicyParameters(autoscalingPolicy.Parameters),
+	}
```
Comment (arthur29) [inline internal/api/handlers/requestadapters/schedulers.go:502]:
```suggestion
        if autoscalingPolicy == nil {
            return nil
        }
        
	return &api.AutoscalingPolicy{
		Type:       string(autoscalingPolicy.Type),
		Parameters: getPolicyParameters(autoscalingPolicy.Parameters),
	}
```

Comment (caiordjesus) [reply internal/api/handlers/requestadapters/schedulers.go:502]: Actually this is impossible because autoscalingPolicy is not a pointer

**Review by arthur29 — APPROVED** [review]
LGTM

**Pull request merged**